x-pack/legacy/plugins/maps/public/layers/fields/es_doc_field.js

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+
+import { AbstractField } from './field';
+import { ESTooltipProperty } from '../tooltips/es_tooltip_property';
+
+export class ESDocField extends AbstractField {
+
+  static type = 'ES_DOC';
+
+  async _getField() {
+    const indexPattern = await this._source.getIndexPattern();
+    return indexPattern.fields.find((field) => field.name === this._fieldName);
```

Comment:
Use `fields.getByName` instead of looping through fields. indexPattern.fields already maps field names so you don't need to search. Search could be expensive since index patterns can have 10000s fields